### DISCUSSION TITLE
Two streams Join Problem

### DISCUSSION THREAD

**Discussion by JackTan25** [risingwavelabs/risingwave#2311] [Q&A] [answered]
In flink,if two streams are running and we need to execute a join with them,flink gives three approaches to solve it,the Window Join,interval Join and Regular Join.So how does risingwave deals with that?

**Comment by TennyZhuang** [comment, marked as answer]
Thanks for your interest in RisingWave!

Currently, RisingWave support Regular join and Window join (generally available).

See https://github.com/singularity-data/risingwave/blob/main/e2e_test/v2/streaming/join.slt for the regular join example.

For WindowJoin, RisingWave support Windowing TVF, then you can do window join like:

```sql
SELECT L.window_start, L.linenumber, L.quantity, O.orderstatus
FROM TUMBLE(lineitems, shipdate, INTERVAL '1' DAYS) as L
JOIN TUMBLE(orders, orderdate, INTERVAL '1' DAYS) as O
ON L.orderkey = O.orderkey and L.window_start = O.window_start;
```

Window table function is implemented recently, and we haven't documented it.

**Reply by JackTan25** [reply]
Could you introduce to me how you store the state?like flink,store them with the join operator together,and in materialize, it's separate and we can share the state store.So how does risingwave do that?

**Reply by TennyZhuang** [reply]
Here is a document about how we store the state in shared storage (currently S3).

We are also working on shared arrangement like materialize, and we will publish our design doc in the near future!

**Reply by JackTan25** [reply]
Ok,I'm waiting
